Display message if recovery key is not valid

infeo opened the pull request:
This PR enhances UX.

It adds two messages to the first reset password dialog: 
* if the recovery key is not valid and
* if the recovery key is valid, but does not belong to this vault.

This change has **security implications**: A user can start guessing the recovery key. We should consider if the increased usability and reduced support justify this.

Screenshots:
![grafik](https://user-images.githubusercontent.com/9036915/219353205-b4fab93b-120a-4e15-8faa-f964c2cb0470.png)
![grafik](https://user-images.githubusercontent.com/9036915/219353281-878a540b-d7da-4e2c-8dbf-d02f416dcb07.png)

tobihagemann commented on src/main/resources/i18n/strings.properties:
```diff
@@ -447,7 +447,9 @@ recoveryKey.display.StorageHints=Keep it somewhere very secure, e.g.:\n • Stor
 recoveryKey.recover.title=Reset Password
 recoveryKey.recover.prompt=Enter your recovery key for "%s":
-recoveryKey.recover.validKey=This is a valid recovery key
+recoveryKey.recover.validKey=This is the valid recovery key
+recoveryKey.recover.wrongKey=This recovery key belongs to a different vault
+recoveryKey.recover.invalidKey=This is not a valid recovery key
```
```suggestion
recoveryKey.recover.correctKey=This recovery key is correct
recoveryKey.recover.wrongKey=This recovery key belongs to a different vault
recoveryKey.recover.invalidKey=This recovery key is not valid
```

tobihagemann commented on src/main/resources/fxml/recoverykey_recover.fxml:
```diff
@@ -25,20 +27,36 @@
+					<FontAwesome5IconView glyph="ANCHOR"/>
+				</graphic>
+			</Label>
+			<Label text="%recoveryKey.recover.validKey" graphicTextGap="6" contentDisplay="LEFT" visible="${(!textarea.text.empty) &amp;&amp; controller.recoveryKeyCorrect}">
```
```suggestion
			<Label text="%recoveryKey.recover.correctKey" graphicTextGap="6" contentDisplay="LEFT" visible="${(!textarea.text.empty) &amp;&amp; controller.recoveryKeyCorrect}">
```

tobihagemann reviewed (changes requested):
> This change has **security implications**: A user can start guessing the recovery key. We should consider if the increased usability and reduced support justify this.

I don't think so. It's like trying to guess the masterkey itself (but less efficient). I'm all for improved usability.

tobihagemann commented on src/main/java/org/cryptomator/ui/recoverykey/RecoveryKeyRecoverController.java:
```diff
@@ -123,23 +142,61 @@ private boolean checkKeyAgainstVaultConfig(byte[] key) {
 
 	/* Getter/Setter */
 
+	public void validateRecoveryKey() {
+		var valid = recoveryKeyFactory.validateRecoveryKey(recoveryKey.get(), unverifiedVaultConfig != null ? this::checkKeyAgainstVaultConfig : null);
+		if (valid) {
+			recoveryKeyState.set(RecoveryKeyState.CORRECT);
+		} else {
+			if (recoveryKeyState.getValue() != RecoveryKeyState.WRONG) { //set via side effect in checkKeyAgainstVaultConfig
+				recoveryKeyState.set(RecoveryKeyState.INVALID);
+			}
+		}
+	}
```
```suggestion
	public void validateRecoveryKey() {
		var valid = recoveryKeyFactory.validateRecoveryKey(recoveryKey.get(), unverifiedVaultConfig != null ? this::checkKeyAgainstVaultConfig : null);
		if (valid) {
			recoveryKeyState.set(RecoveryKeyState.CORRECT);
		} else if (recoveryKeyState.getValue() != RecoveryKeyState.WRONG) { // set via side effect in checkKeyAgainstVaultConfig()
			recoveryKeyState.set(RecoveryKeyState.INVALID);
		}
	}

	/* Getter/Setter */
```

Just a minor optimization in the if/else statement. But also moved the getter/setter comment down. I guess it's not really a getter/setter in the sense of JavaFX?

tobihagemann reviewed (changes requested):
Furthermore: Are you sure that this won't mess up translations since you've now introduced a _new_ key for every language code (instead of pulling the translations from Crowdin)?

infeo commented:
> Are you sure that this won't mess up translations since you've now introduced a new key for every language code

I#M not sure. I reverted that change.
